src/components/PostDetails.tsx

```diff
@@ -1,117 +1,136 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { deleteComment, getComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | undefined,
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [commentsIsLoading, setCommentsIsLoading] = useState(false);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [commentsLoadingError, setCommentsLoadingError] = useState(false);
+  const [addCommentFormVisible, setAddCommentFormVisible] = useState(false);
+
+  useEffect(() => {
+    setAddCommentFormVisible(false);
+
+    if (selectedPost) {
+      setCommentsIsLoading(true);
+      setCommentsLoadingError(false);
+      getComments(selectedPost.id)
+        .then(data => setComments(data))
+        .catch(() => setCommentsLoadingError(true))
+        .finally(() => setCommentsIsLoading(false));
+    }
+  }, [selectedPost]);
+
+  const addNewComment = (newComment: Comment) => {
+    setComments(prev => [...prev, newComment]);
+  };
+
+  const deleteCommentHandler = (commentId:number) => {
+    setComments(comments.filter(el => el.id !== commentId));
+    deleteComment(commentId)
+      .then(() => {})
+      .catch(() => {})
+      .finally();
+  };
```

Comment:
`catch` is enough in case when you didn't expect to receive anything
```suggestion
const deleteCommentHandler = (commentId: number) => {
  setComments(comments.filter((el) => el.id !== commentId));
  deleteComment(commentId).catch((e) => {
    throw new Error(`error deleting todos: ${(e as any)?.message}`);
  });
};
```